addons/account_facturx/models/account_move.py

```diff
@@ -224,97 +228,15 @@ def _import_facturx_invoice(self, tree):
-                    for filename_obj, content_obj in list(zip(embedded_files, embedded_files[1:]))[::2]:
-                        content = content_obj.getObject()['/EF']['/F'].getData()
-
-                        if filename_obj == 'factur-x.xml':
-                            try:
-                                tree = etree.fromstring(content)
-                            except Exception:
-                                continue
-
-                            self._import_facturx_invoice(tree)
-                            self._remove_ocr_option()
-                            buffer.close()
-            except except_orm as e:
-                raise e
-            except Exception as e:
-                # Malformed pdf
-                _logger.exception(e)
+    def _detect_facturx(self, tree, file_name):
+        return {
+            'flag': tree.tag == '{urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100}CrossIndustryInvoice',
+            'error': None,
+        }
 
     @api.model
     def _get_xml_decoders(self):
-        ''' List of usable decoders to extract invoice from attachments.
-
-        :return: a list of triplet (xml_type, check_func, decode_func)
-            * xml_type: The format name, e.g 'UBL 2.1'
-            * check_func: A function taking an etree and a file name as parameter and returning a dict:
-                * flag: The etree is part of this format.
-                * error: Error message.
-            * decode_func: A function taking an etree as parameter and returning an invoice record.
-        '''
-        # TO BE OVERWRITTEN
-        return []
-
-    def _create_invoice_from_xml(self, attachment):
-        decoders = self._get_xml_decoders()
-
-        # Convert attachment -> etree
-        content = base64.b64decode(attachment.datas)
-        try:
-            tree = etree.fromstring(content)
-        except Exception:
-            _logger.exception('The xml file is badly formatted : {}'.format(attachment.name))
-
-        for xml_type, check_func, decode_func in decoders:
-            check_res = check_func(tree, attachment.name)
-
-            if check_res.get('flag') and not check_res.get('error'):
-                invoice_ids = decode_func(tree)
-                if invoice_ids:
-                    invoice_ids._remove_ocr_option()
-                    break
-
-        try:
-            return invoice_ids
-        except UnboundLocalError:
-            _logger.exception('No decoder was found for the xml file: {}. The file is badly formatted, not supported or the decoder is not installed'.format(attachment.name))
-
-    def _remove_ocr_option(self):
-        if 'extract_state' in self:
-            self.write({'extract_state': 'done'})
+        # OVERRIDE
+        ubl_decoders = [('Factur-X', self._detect_facturx, self._import_facturx_invoice, self._export_as_facturx_xml, True)]
+        return super(AccountMove, self)._get_xml_decoders() + ubl_decoders
```

Comment:
You know, the size of this kind of sentence could be reduced: `return super()._get_xml_decoders() + ubl_decoders`